src/comp/shareButtonData/data.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  faGetPocket,
+  faLine,
+  faTwitter,
+  faFacebookF,
+} from "@fortawesome/free-brands-svg-icons";
+import {
+  faEnvelope,
+  faLink,
+  faShareAlt,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+
+export type Icon = {
+  Twitter: IconDefinition;
+  Facebook: IconDefinition;
+  LINE: IconDefinition;
+  Mail: IconDefinition;
+  Pocket: IconDefinition;
+  Copy: IconDefinition;
+};
+export type IconUrl = {
+  Twitter: string;
+  Facebook: string;
+  LINE: string;
+  Mail: string;
+  Pocket: string;
+  Copy: string;
+};
+export type IconName = Array<keyof Icon>;
+
+export const mainIconName: IconName = ["Twitter", "Facebook"];
+export const subIconName: IconName = ["LINE", "Mail", "Pocket"];
+export const icons: Icon = {
+  Twitter: faTwitter,
+  Facebook: faFacebookF,
+  LINE: faLine,
+  Mail: faEnvelope,
+  Pocket: faGetPocket,
+  Copy: faLink,
+};
+
+type UrlProps = {
+  url: string;
+  introduction: string;
+  title: string;
+};
+export const Url = (props: UrlProps) => {
+  const sentenceLengthLimits = 40;
+  const intro = props.introduction.substr(0, sentenceLengthLimits);
```

Comment:
```suggestion
  const intro = props.introduction.substr(0, 40);
```
一回しか使われてないのと、これだけでも十分伝わるので変数にする必要は無いと思います

Reply:
他の所で同じようにしてるから変数にしたほうがいいのかなって思ったんだけど、

Reply:
私はリテラルに名前をつける目的で定数にしておくのは賛成ですね

Reply:
あー…無理に短くする必要は確かにないか
わかりました、これはこのままで